### PR TITLE
NEW: Add option MAIN_ADD_LINE_AT_POSITION to choose position of line during insert

### PR DESCRIPTION
**Pull request by atm-gauthier**
When MAIN_VIEW_LINE_NUMBER is enabled, we can now choose in which position we want to add a new line (available on propal, order, invoice, propal fourn, order fourn, invoice fourn)

We select position :
image

And then it's added on selected line
image

By default, lines are added at the end as well as current workflow :
image

### CODE REVIEW

**Comment by atm-gauthier** [comment]
Hi @eldy, for my feature the "for" loop seems to be the most appropriate but i get errors like this :

image

What's your opinion about that ?

**Comment by atm-maxime** [comment]
Suggestions :
Use an input type number instead of a drop-down, because if your document has 100 lines, a drop-down will be hard to use.
The keyword OnLine for the translation might be wrongly understood, use AddLineOnPosition instead.

**Comment by atm-gauthier** [comment]
@atm-maxime i agree with your proposition, i'm just waiting for @eldy feedback.
If he is ok with this feature, i will replace select by input number

**Comment by atm-gauthier** [comment]
Hi @eldy, i finally did what @atm-maxime said, it's now an input number :

image

File: htdocs/comm/propal/class/propal.class.php [hunk 1]
```diff
@@ -734,6 +735,10 @@ public function addline($desc, $pu_ht, $qty, $txtva, $txlocaltax1 = 0.0, $txloca
 				// Reorder if child line
 				if (!empty($fk_parent_line)) {
 					$this->line_order(true, 'DESC');
+				} elseif ($ranktouse > 0 && $ranktouse <= count($this->lines)) { // Update all rank of all other lines
+					for ($ii = $ranktouse; $ii <= count($this->lines); $ii++) {
```
Comment (frederic34) [inline htdocs/comm/propal/class/propal.class.php:739]: `$linecount = count($this->lines);
for ($ii = $ranktouse; $ii <= $linecount; $ii++) {`

Comment (atm-gauthier) [reply htdocs/comm/propal/class/propal.class.php]: @frederic34 you were right, it's ok now with my PR, thank you very much ! :)

**Comment by atm-gauthier** [comment]
I did a new version with input number :

image

File: htdocs/core/tpl/objectline_create.tpl.php [hunk 2]
```diff
@@ -309,6 +310,11 @@
 			echo '<input type="hidden" name="pbq" id="pbq" value="">';
 			echo '</span>';
 		}
+
+		if (!empty($conf->global->MAIN_VIEW_LINE_NUMBER)) {
```
Comment (eldy) [inline htdocs/core/tpl/objectline_create.tpl.php:314]: I would suggest to use another constant like
MAIN_ADD_LINE_NUMBER
and keep MAIN_VIEW_LINE_NUMBER for current behaviour (read only the column)

Comment (atm-gauthier) [reply htdocs/core/tpl/objectline_create.tpl.php]: Hi @eldy, i did it and created hidden conf "MAIN_ADD_LINE_AT_POSITION"

**Comment by atm-gauthier** [comment]
@eldy i did modification you asked, can you merge this PR ?
I often have to re-merge develop in my branch... ;)

**Pull request merged**